### PR TITLE
Added net6.0-windows on WPF and WinForms and workaround MSBuild.Sdk.Extras issue

### PR DESCRIPTION
**Pull request by jeremyVignelles**
### Description of Change ###

- Added net6.0-windows TFM on LibVLCSharp.WinForms
- Added net6.0-windows TFM on LibVLCSharp.WPF
- Added `DESKTOP` constant to replace the ([flaky](https://github.com/novotnyllc/MSBuildSdkExtras/issues/279)) `NETFRAMEWORK || NETSTANDARD || NET6_0`

### Issues Resolved ### 
Users referencing LVS.WinForms in .net6.0 should have a working project now

### API Changes ###
 
None (Fixes missing APIs in 3.6.3)

### Platforms Affected ### 

- Core (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [X] Rebased on top of the target branch at time of PR
- [X] Changes adhere to coding standard

### CODE REVIEW

**Comment by jeremyVignelles** [comment]
> Replacing `MediaPlayer` by `Shared.MediaPlayer` fixes it (but that should not be needed)

Is there any namespace somewhere that's named MediaPlayer and that conflicts with the type ? Do you have more info on this ? why did that not break before ? (different SDK ?)

File: samples/LibVLCSharp.NetCore.Sample/LibVLCSharp.NetCore.Sample.csproj [hunk 1]
```diff
@@ -1,10 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework Condition="$([MSBuild]::IsOsPlatform('Windows'))">net6.0</TargetFramework>
+    <TargetFrameworks Condition="$([MSBuild]::IsOsPlatform('Windows'))">net6.0;net6.0-windows;net6.0-macos</TargetFrameworks>
     <TargetFramework Condition="!$([MSBuild]::IsOsPlatform('Windows'))">netcoreapp3.1</TargetFramework>
-    <Platforms>AnyCPU;x64;x86</Platforms>
```
Comment (jeremyVignelles) [inline samples/LibVLCSharp.NetCore.Sample/LibVLCSharp.NetCore.Sample.csproj:6]: not sure we should remove these, a `net6.0-windows` will by default be AnyCPU, but what if you want to test the `x64` and `x86` loading ?

If the goal is to test the initialization of the core in the cross-product of all platforms combination, this should be kept.

Comment (mfkl) [reply samples/LibVLCSharp.NetCore.Sample/LibVLCSharp.NetCore.Sample.csproj:6]: Sure

**Comment by mfkl** [comment]
> > Replacing `MediaPlayer` by `Shared.MediaPlayer` fixes it (but that should not be needed)
> 
> Is there any namespace somewhere that's named MediaPlayer and that conflicts with the type ? Do you have more info on this ? why did that not break before ? (different SDK ?)

SDK issue I'd say. Removing the macos target fixes it (though the build error does not prevent the DLL to be built).

```diff
-   <TargetFrameworks Condition="$([MSBuild]::IsOsPlatform('Windows'))">net6.0;net6.0-windows;net6.0-macos</TargetFrameworks>
+   <TargetFrameworks Condition="$([MSBuild]::IsOsPlatform('Windows'))">net6.0;net6.0-windows</TargetFrameworks>
```

**Comment by mfkl** [comment]
For WinForms and WPF samples, this change is required for net6.0 (breaks legacy TFMs)
```diff
- <Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+ <Project Sdk="Microsoft.NET.Sdk">
```

Cross-targetting for our "internal" samples project seems like a lot of overhead now. Maybe we should just keep net6.0 / latest for these test samples?

**Pull request merged**